strawberryfields/gbs/similarity.py

```diff
@@ -428,3 +430,53 @@ def feature_vector_sampling(
     count = Counter(e)
 
     return [count[p] / n_samples for p in event_photon_numbers]
+
+
+def feature_vector_mc(
+    graph: nx.Graph,
+    event_photon_numbers: list,
+    max_count_per_mode: int = 2,
+    n_mean: int = 5,
+    samples: int = 1000,
+) -> list:
+    r"""Calculates feature vector using a Monte Carlo estimation of event probabilities.
+
+    The feature vector is composed of event probabilities :math:`p_{k}` with all events
+    :math:`E_{k}` having a maximum photon count in each mode of ``max_count_per_mode``. Events
+    are specified by their total photon number :math:`k` and those chosen as part of the feature
+    vector can be specified through :math:`\mathbf{k}` (using the ``event_photon_numbers``
+    argument). The resultant feature vector is
+
+    .. math::
+        f_{\mathbf{k}} = (p_{k_{1}}, p_{k_{2}}, \ldots)
+
+    Probabilities are reconstructed using Monte Carlo estimation.

```

Comment:
From what I've tested so far, it takes tens of seconds to compute events up to 8 photons. I would advise that we wait until testing applications further before adding warnings since they may not be needed